### ISSUE TITLE
Search for the Korath: mission failed

### ISSUE THREAD

**Issue by waterhouse** [endless-sky/endless-sky#5908]
**Describe the bug**
In the "Search for the Korath" mission (where you find a hostile Korath fleet in Aescolanus), if you kill a single Korath ship (including a fighter), you get a "Mission failed".

It seems that this is not supposed to be a "failure"—you would get almost identical text upon landing, but with an additional "You have demonstrated good initiative, however I suspect that it was not the only hunter fleet out there."

However, as a player, I assumed it *was* a bad thing (one that would prevent my progressing further in the string), and did not get far enough to find out otherwise—I immediately raised an eyebrow, frowned, reloaded and retried until I got through it without "failing".  I expect this reaction to be fairly common.

I note further that the "not the only hunter fleet" text suggests the player destroyed the whole fleet, yet that text happens if the player destroys just a single ship.

**To Reproduce**
1. Accept the "Search for the Korath" mission.
2. Go to Aescolanus.
3. Kill a single ship.
4. Observe the "Mission failed".

**Expected behavior**
I would expect that killing any or all of the Korath ships (which are hostile, and the whole point of the operation is to kill them anyway—the very next mission is "Destroy Korath") would not cause a "Mission failed".

Also, if we do have the "killing -> extra text" (which does make sense, to tell a player who's killed the fleet why the next mission needs to happen), then that should appear only when the player has killed the *whole* fleet.

I imagine the usual way to do this would be to make "Search" start a silent mission B whose goal is to kill the fleet (perhaps make B auto-fail itself when that goal is accomplished), to make "Search" show conditional text depending on whether B had been finished (or auto-failed), and to make "Search" abort B at the end.  (And "Destroy Korath"'s conditional text would need to be kept abreast of these changes.)

**Version**
599142ac9a1ec88dc1ee525c1c776ac29f8ee9ac (Apr 17)

**Comment by Zitchas** [comment]
This is an instance of the game mechanics don't really support the intended behaviour. The mission is only capable of a binary result. Namely, either none of the ships were damaged, in which case you succeed, or any of the ships were damaged, in which case you "failed." Note that this is only a failure in as much as you didn't meet the succeed conditions. It ***does not actually fail the story, just the mechanical mission***. You can continue the story just fine, with whether or not that mission was "failed" or "completed" resulting in a change in dialog to reflect how you handled the Korath. 

Now that I'm thinking about it, I could theoretically use a whole bunch of different missions to each spawn one of the Korath ships, and then test for how many were killed to adjust the text, but at that point it seems like code bloat for a situation where - let's be honest - if the player is killing any Korath, they are *probably* killing all or at least most of them.

**Comment by waterhouse** [comment]
> theoretically use a whole bunch of different missions to each spawn one of the Korath ships, and then test for how many were killed to adjust the text

I think it's *slightly* better than that.  I haven't read the mission coding manual, and haven't tested the below code, but it seems a mission can have multiple "npc kill" clauses; and it seems that, within those clauses, at least "dialog" clauses execute immediately when the last ship is killed; so if "++" clauses do too, then you could do something like:
```
        npc kill
                government "Korath"
                personality staying surveillance
                system "Aescolanus"
                fleet "Large Remnant"
                "ships killed" ++
        npc kill
                government "Korath"
                personality staying surveillance
                system "Aescolanus"
                fleet "Large Remnant"
                "ships killed" ++
        npc kill
                government "Korath"
                personality staying surveillance
                system "Aescolanus"
                fleet "Large Remnant"
                "ships killed" ++
...
```
So O(n) code but only O(1) missions to do that.
Then, for the debriefing—I dunno if you can substitute integers into text (EV Nova had things like `{[boolean] "true-text" "false-text"}`, but I don't recall it being able to insert integers).  I also don't know if it would be possible to use an expression like `5 - "ships killed"` as the number of instances of a fleet.
Probably still too much code, though.

> The main problem here is that the "mission failed" text displays, when the mission is not actually a fail.

Agreed.

> Some way to suppress that notification for a mission is desirable.

Well, if a mission is marked invisible, then that suppresses the notification.  So you can do it using a second invisible mission, possibly one whose "npc kill" clause will fail it (if you want the condition to be "if the player kills the whole fleet").

**Comment by Amazinite** [comment]
Duplicate of #5246.

**Comment by waterhouse** [comment]
On a side note:

> if the player is killing any Korath, they are probably killing all or at least most of them.

For the heavy ships, I'd agree.  Not for the fighters, though: in my case, although I was trying to avoid confrontation, I couldn't stop my 6 escorts from killing at least one fighter (incidentally, I was doing an "observe the electrical storms" mission as well, so I had to stay in the system for a bit) even after a couple of tries.  I ultimately resorted to making my escorts stay home while I entered the system alone and cloaked.
